goagen/gen_test/generator.go

```diff
@@ -0,0 +1,299 @@
+package gentest
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"text/template"
+
+	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/goagen/codegen"
+	"github.com/goadesign/goa/goagen/utils"
+	"github.com/spf13/cobra"
+)
+
+// Generator is the application code generator.
+type Generator struct {
+	genfiles []string
+}
+
+// Generate is the generator entry point called by the meta generator.
+func Generate() (files []string, err error) {
+	api := design.Design
+	if err != nil {
+		return nil, err
+	}
+	g := new(Generator)
+	root := &cobra.Command{
+		Use:   "goagen",
+		Short: "Test generator",
+		Long:  "controller test and package generator",
```

Comment:
a bit clunky perhaps, but the idea was; it generates the test helpers and the 'test' folder.